cpp/tests/copying/scatter_struct_tests.cpp

```diff
@@ -241,3 +241,21 @@ TYPED_TEST(TypedStructScatterTest, SourceSmallerThanTargetScatterTest)
   CUDF_TEST_EXPECT_COLUMNS_EQUAL(
     *structs_expected, scatter_structs(structs_src, structs_tgt, scatter_map), print_all);
 }
+
+TYPED_TEST(TypedStructScatterTest, IntStructNullMaskRegression)
+{
+  auto child_0      = int32s_col({0, -1, 2}, null_at(1));
+  auto struct_col_0 = structs_col({child_0}).release();
+
+  auto child_1      = int32s_col({20});
+  auto struct_col_1 = structs_col({child_1}).release();
+
+  auto scatter_map = int32s_col{2}.release();
+
+  auto expected_child  = int32s_col({0, -1, 20}, null_at(1));
+  auto expected_struct = structs_col({expected_child}).release();
+
+  auto result_ptr = scatter_structs(struct_col_1, struct_col_0, scatter_map);
+
+  CUDF_TEST_EXPECT_COLUMNS_EQUAL(*expected_struct, result_ptr, print_all);
```

Comment:
```suggestion
  auto child_tgt      = col_wrapper({0, -1, 2}, null_at(1));
  auto struct_col_tgt = structs_col({child_tgt}).release();

  auto child_src      = col_wrapper({20});
  auto struct_col_src = structs_col({child_src}).release();

  auto scatter_map = int32s_col{2}.release();

  auto expected_child  = col_wrapper({0, -1, 20}, null_at(1));
  auto expected_struct = structs_col({expected_child}).release();

  auto const result = scatter_structs(struct_col_src, struct_col_tgt, scatter_map);
  CUDF_TEST_EXPECT_COLUMNS_EQUAL(*expected_struct, result, print_all);
```

Reply:
This suggestion does not compile

Reply:
What's the error message?

Reply:
```
../tests/copying/scatter_struct_tests.cpp: In member function ‘virtual void TypedStructScatterTest_IntStructNullMaskRegression_Test<gtest_TypeParam_>::TestBody()’:
../tests/copying/scatter_struct_tests.cpp:247:27: error: there are no arguments to ‘col_wrapper’ that depend on a template parameter, so a declaration of ‘col_wrapper’ must be available [-fpermissive]
  247 |     auto child_tgt      = col_wrapper({0, null, 2}, null_at(1));
      |                           ^~~~~~~~~~~
../tests/copying/scatter_struct_tests.cpp:247:27: note: (if you use ‘-fpermissive’, G++ will accept your code, but allowing the use of an undeclared name is deprecated)
../tests/copying/scatter_struct_tests.cpp:250:25: error: ‘col_wrapper’ was not declared in this scope
  250 |   auto child_src      = col_wrapper{20};
      |                         ^~~~~~~~~~~
../tests/copying/scatter_struct_tests.cpp: In instantiation of ‘void TypedStructScatterTest_IntStructNullMaskRegression_Test<gtest_TypeParam_>::TestBody() [with gtest_TypeParam_ = unsigned int]’:
../tests/copying/scatter_struct_tests.cpp:245:1:   required from here
../tests/copying/scatter_struct_tests.cpp:247:38: error: ‘col_wrapper’ was not declared in this scope
  247 |     auto child_tgt      = col_wrapper({0, null, 2}, null_at(1));
  ....
```

Reply:
I see. Just fixed that (edited the code above). Please test again.

Reply:
This worked, thanks @ttnghia